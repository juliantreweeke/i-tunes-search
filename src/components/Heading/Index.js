import PropTypes from 'prop-types';
import React from 'react';
import styles from './heading.module.css';


const Heading = ({ children, headingLevel, variant, size }) => {

  const validHeadingLevels = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];

  const Title = validHeadingLevels.includes(headingLevel) ? headingLevel : 'p'; 

  return (
    <Title className={styles.heading} data-variant={variant} data-size={size}>{children}</Title>
  );
};

Heading.propTypes = {
  children: PropTypes.node.isRequired,
  headingLevel: PropTypes.string.isRequired,
  variant: PropTypes.string,
};

export default Heading;
