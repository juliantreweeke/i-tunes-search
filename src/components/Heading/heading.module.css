.heading {
  color: var(--color-text-dark);
  transition: all 0.45s ease-in-out;
  animation: fadeIn 1s steps(80, end);
  max-width: 647px;
  margin-left: auto;
  margin-right: auto;
  font-size: 2.4rem;
}

.heading[data-variant="light"] {
  color: var(--color-text-light);
}

.heading {
  font-size: var(--text-lg);
}

.heading[data-size="large"] {
  font-size: var(--text-xl);
}

.heading[data-size="medium"] {
  font-size: var(--text-md);
}

.heading[data-size="small"] {
  font-size: var(--text-sm);
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 100;
  }
}
