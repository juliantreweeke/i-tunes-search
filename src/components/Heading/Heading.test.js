import React from "react";
import { render } from "@testing-library/react";
import Heading from "./Index";

describe("Heading", () => {
  it("should render", () => {
    const { container } = render(<Heading headingLevel="h1" />);
    expect(container).toBeTruthy();
  });

  it("matches snapshot with prop data", () => {
    const { container } = render(<Heading headingLevel="h1" />);
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with invalid headingLevel prop data", () => {
    const { container } = render(<Heading headingLevel="invalidvalue" />);
    expect(container).toMatchSnapshot();
  });
});
