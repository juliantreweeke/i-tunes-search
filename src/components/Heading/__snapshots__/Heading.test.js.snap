// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading matches snapshot with invalid headingLevel prop data 1`] = `
<div>
  <p
    class="heading"
  />
</div>
`;

exports[`Heading matches snapshot with prop data 1`] = `
<div>
  <h1
    class="heading"
  />
</div>
`;
